src/Agent.Listener/Configuration/NativeWindowsServiceHelper.cs

```diff
@@ -789,6 +795,26 @@ private byte[] GetSidBinaryFromWindows(string domain, string user)
             }
         }
 
+        public bool IsTheSameUserLoggedIn(string domainName, string userName)
```

Comment:
Or "TestActiveSessionExists" if you want to stick with .NET convention that methods are verbs.